### ISSUE TITLE
Bug: `--feature=swift.use_global_index_store` can error while reading units

### ISSUE THREAD

**Issue by brentleyjones** [bazelbuild/rules_swift#907]
```
error: failed to read unit file /BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+Partial__SPACE__MutableCollection.swift.o-3PKKL4WGTUWG9 -- Failed opening '/BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+Partial__SPACE__MutableCollection.swift.o-3PKKL4WGTUWG9': No such file or directory
error: failed to read unit file /BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+Partial__SPACE__RangeReplaceableCollection.swift.o-2SYSC34735DB4 -- Failed opening '/BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+Partial__SPACE__RangeReplaceableCollection.swift.o-2SYSC34735DB4': No such file or directory
error: failed to read unit file /BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+Partial__SPACE__SetAlgebra+Basics.swift.o-3F9KMIRYW78UI -- Failed opening '/BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+Partial__SPACE__SetAlgebra+Basics.swift.o-3F9KMIRYW78UI': No such file or directory
error: failed to read unit file /BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+Partial__SPACE__SetAlgebra+Operations.swift.o-1FR9D99EY6S6R -- Failed opening '/BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+Partial__SPACE__SetAlgebra+Operations.swift.o-1FR9D99EY6S6R': No such file or directory
error: failed to read unit file /BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+Partial__SPACE__SetAlgebra+Predicates.swift.o-CGMTK2713X0Q -- Failed opening '/BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+Partial__SPACE__SetAlgebra+Predicates.swift.o-CGMTK2713X0Q': No such file or directory
error: failed to read unit file /BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+RandomAccessCollection.swift.o-RBRJKLFLLPKG -- Failed opening '/BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+RandomAccessCollection.swift.o-RBRJKLFLLPKG': No such file or directory
error: failed to read unit file /BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+ReserveCapacity.swift.o-15QXFQO7P86K1 -- Failed opening '/BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+ReserveCapacity.swift.o-15QXFQO7P86K1': No such file or directory
error: failed to read unit file /BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+SubSequence.swift.o-7YPMI03P9MR7 -- Failed opening '/BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+SubSequence.swift.o-7YPMI03P9MR7': No such file or directory
error: failed to read unit file /BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+Testing.swift.o-30JGQKDYTRLWV -- Failed opening '/BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+Testing.swift.o-30JGQKDYTRLWV': No such file or directory
error: failed to read unit file /BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+UnorderedView.swift.o-3NPSWABCIFDFV -- Failed opening '/BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+UnorderedView.swift.o-3NPSWABCIFDFV': No such file or directory
error: failed to read unit file /BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+UnstableInternals.swift.o-N3G3UE5WW4IY -- Failed opening '/BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet+UnstableInternals.swift.o-N3G3UE5WW4IY': No such file or directory
error: failed to read unit file /BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet.swift.o-1XBG26ZB0RM4X -- Failed opening '/BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/OrderedSet.swift.o-1XBG26ZB0RM4X': No such file or directory
error: failed to read unit file /BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/RandomAccessCollection+Offsets.swift.o-1FR4JB101EBP0 -- Failed opening '/BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/RandomAccessCollection+Offsets.swift.o-1FR4JB101EBP0': No such file or directory
error: failed to read unit file /BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/_UnsafeBitset.swift.o-14C13QMHG2DA1 -- Failed opening '/BAZEL_EXECUTION_ROOT/bazel-out/_global_index_store/v5/units/_UnsafeBitset.swift.o-14C13QMHG2DA1': No such file or directory
```

**Comment by brentleyjones** [comment]
@jerrymarino Have you seen this before? Do you know what would cause it/how to work around it?

**Comment by jpsim** [comment]
It seems to be caused when `swift.use_global_index_store` is used with worker sandboxing (`--worker_max_instances=HOST_CPUS --worker_sandboxing=true`).

**Comment by brentleyjones** [comment]
I believe this is fixes in later versions of rules_swift. I could reproduce on 1.1.0, but not on later versions.

**Comment by jpsim** [comment]
I'm still seeing this with 6f13e3a8e9d48b8e3e70732bf34c677c3dc2a5e2

Setting `--features=-swift.use_global_index_store` works around the issue.

**Comment by brentleyjones** [comment]
Can you test with https://github.com/bazelbuild/rules_swift/commit/2918678907af3bc63ef00ed66b71f1d762542e0b? I think I fixed it there.

**Comment by jpsim** [comment]
Appears fixed with 2918678907af3bc63ef00ed66b71f1d762542e0b, thanks!
